src/Tempest/Http/src/Routing/Matching/GenericRouteMatcher.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http\Routing\Matching;
+
+use Psr\Http\Message\ServerRequestInterface as PsrRequest;
+use Tempest\Http\MatchedRoute;
+use Tempest\Http\Route;
+use Tempest\Http\RouteConfig;
+use Tempest\Http\Routing\Construction\MarkedRoute;
+
+final readonly class GenericRouteMatcher implements RouteMatcher
+{
+    public function __construct(private RouteConfig $routeConfig)
+    {
+    }
+
+    public function match(PsrRequest $request): ?MatchedRoute
+    {
+        // Try to match routes without any parameters
+        if (($staticRoute = $this->matchStaticRoute($request)) !== null) {
+            return $staticRoute;
+        }
+
+        // match dynamic routes
+        return $this->matchDynamicRoute($request);
+    }
+
+    private function matchStaticRoute(PsrRequest $request): ?MatchedRoute
+    {
+        $staticRoute = $this->routeConfig->staticRoutes[$request->getMethod()][$request->getUri()->getPath()] ?? null;
```

Comment:
So, technically the static routes are in the tree as well, but we don't need them? That makes me wonder:

1. why do we keep them in the tree?
2. is it again a code smell indicating that we're doing two related things in separate places?
3. FTR: performance would be a good argument to keep them apart, but if that is the case, I think we can get rid of static routes in the tree alltogether?

Reply:
Don't know why I thought they were 🫣 sorry for the noise